pkg/reconciler/v1alpha1/templating/templating.go

```diff
@@ -0,0 +1,71 @@
+/*
+ Copyright 2019 Knative Authors LLC
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+     http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package templating
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/knative/pkg/apis"
+)
+
+const parameterSubstitution = "[_a-zA-Z][_a-zA-Z0-9.-]*"
+
+func ValidateVariable(name, value, prefix, contextPrefix, locationName, path string, vars map[string]struct{}) *apis.FieldError {
```

Comment:
now that we have this function in it's own package, would it make sense to have some unit tests for it specifically? :innocent:

Reply:
...yes. It would. I'll do that after the working group meeting. =)

Reply:
And done!